pkgs/development/compilers/edk2/default.nix

```diff
@@ -50,13 +66,17 @@ edk2 = stdenv.mkDerivation {
     homepage = https://sourceforge.net/projects/edk2/;
     license = stdenv.lib.licenses.bsd2;
     branch = "UDK2017";
-    platforms = ["x86_64-linux" "i686-linux" "aarch64-linux"];
+    platforms = ["x86_64-linux" "i686-linux" "aarch64-linux" "armv7l-linux"];
   };
 
   passthru = {
+    inherit targetArch hostArch;
     setup = projectDscPath: attrs: {
-      buildInputs = [ pythonEnv ] ++
-        stdenv.lib.optionals (attrs ? buildInputs) attrs.buildInputs;
+      buildInputs = stdenv.lib.optionals (attrs ? buildInputs) attrs.buildInputs;
+      nativeBuildInputs = [ buildPythonEnv ] ++
+        stdenv.lib.optionals (attrs ? nativeBuildInputs) attrs.nativeBuildInputs;
+
+      depsBuildBuild = [ buildPackages.iasl ];
```

Comment:
Let's start slowly. What's the difference between `buildPackages.buildPackages.iasl` and `buildPackages.iasl`? Maybe it's related to the whole target/build/host platforms which I still don't grok entirely.

`iasl` here is a specialized compiler. It is an ACPI compiler.

I'm not even sure why I have put this into `depsBuildBuild` instead of `nativeBuildInputs`. Re-reading `depsBuildBuild`'s description I am still unsure why it would be used. But re-reading `nativeBuildInputs` I'm thinking it should have been `nativeBuildInputs`.

(I'll update this soon with cleanups, once done I'll re-ask you to take a look.)

Reply:
Well, Is ISL used at build time or at run time? I would think it would be used by the built edk2, so at run time, but maybe it is just used to build some standard library thing so used at build time?

Reply:
Take a look at slide 12: https://docs.google.com/presentation/d/11B3Igxj0KmsxgT_UQvKufd5El_oH-T__Sl3JNCZeOnY/edit#slide=id.g4bf2104f71_0_46

Reply:
From understanding this belongs in `nativeBuildInputs` as it is target independent byte-code it produces: https://wiki.osdev.org/AML